### PR TITLE
[in_app_purchase] Add play country code api

### PR DESCRIPTION
**Pull request by reidbaker**
- Initial add of country code, no tests
- Add tests for getBillingConfig

fixes flutter/flutter/issues/141627

Hold merging on: 
- [x] End to end manual test using https://codelabs.developers.google.com/codelabs/flutter-in-app-purchases#0 

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

### CODE REVIEW

File: packages/in_app_purchase/in_app_purchase_android/lib/src/in_app_purchase_android_platform_addition.dart [hunk 1]
```diff
@@ -146,4 +146,15 @@ class InAppPurchaseAndroidPlatformAddition
       (BillingClient client) => client.isFeatureSupported(feature),
     );
   }
+
+  /// Returns Play billing country code based on ISO-3166-1 alpha2 format.
+  ///
+  /// See: https://developer.android.com/reference/com/android/billingclient/api/BillingConfig
+  /// See: https://unicode.org/cldr/charts/latest/supplemental/territory_containment_un_m_49.html
+  Future<String> getBillingConfig() async {
+    return _billingClientManager.runWithClientNonRetryable(
```
Comment (gmackall) [inline packages/in_app_purchase/in_app_purchase_android/lib/src/in_app_purchase_android_platform_addition.dart:155]: Should this return a `BillingConfigWrapper` and then be also be retryable, as it returns a subclass of `HasBillingResponse`? 
https://github.com/flutter/packages/blob/cbe810031b883fcb7f602c8ecf8af05b7cf75905/packages/in_app_purchase/in_app_purchase_android/lib/src/billing_client_wrappers/billing_client_manager.dart#L92

Comment (reidbaker) [reply packages/in_app_purchase/in_app_purchase_android/lib/src/in_app_purchase_android_platform_addition.dart:155]: I wanted to keep the exposed api as small as possible so I thought the country code was the right level to expose. I didnt even think about if it should be retryable I just mirrored the other commands. Let me look at what that does and make a call.

**Comment by reidbaker** [comment]
No changelog needed for the ios readme change. 
Changelog is required for android change. 
https://discord.com/channels/608014603317936148/608020293944082452/1201922969388187721

File: packages/in_app_purchase/in_app_purchase_android/lib/src/billing_client_wrappers/billing_client_wrapper.dart [hunk 2]
```diff
@@ -324,6 +325,21 @@ class BillingClient {
     return result ?? false;
   }
 
+  /// BillingConfig method channel string identifier.
```
Comment (mossmana) [inline packages/in_app_purchase/in_app_purchase_android/lib/src/billing_client_wrappers/billing_client_wrapper.dart:328]: Nit: Did you want to match the comment style for all lines? i.e. consistently use either /// on this line vs // on the next line

File: packages/in_app_purchase/in_app_purchase_android/CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,6 @@
-## NEXT
+## 0.3.0+18
 
+* Adds country code api for android.
```
Comment (gmackall) [inline packages/in_app_purchase/in_app_purchase_android/CHANGELOG.md:3]: nit: specify the api, and also given that this is the android changelog maybe remove "for android"

Something like "Adds new `getCountryCode()` method to `InAppPurchaseAndroidPlatformAddition` to get a customer's country code."

File: packages/in_app_purchase/in_app_purchase_android/lib/src/billing_client_wrappers/billing_client_wrapper.dart (hunk 2 above)
Comment (reidbaker) [reply packages/in_app_purchase/in_app_purchase_android/lib/src/billing_client_wrappers/billing_client_wrapper.dart:328]: /// Is used for dart doc 
// is used for comments. In this case I want callers to see the contends in /// and maintainers to see the comments after //. 

If you want I can move the developer comments inside the function

File: packages/in_app_purchase/in_app_purchase_android/example/README.md [hunk 4]
```diff
@@ -7,3 +7,57 @@ package.
...
+4. Update `APP_ID` in `example/android/app/build.gradle` to match your package
+   ID in the PDC.
+
+5. Create an `example/android/keystore.properties` file with all your signing
+   information. `keystore.example.properties` exists as an example to follow.
+   It's impossible to use any of the `BillingClient` APIs from an unsigned APK.
+   See
+   [here](https://developer.android.com/studio/publish/app-signing#secure-shared-keystore)
+   and [here](https://developer.android.com/studio/publish/app-signing#sign-apk)
+   for more information.
```
Comment (gmackall) [inline packages/in_app_purchase/in_app_purchase_android/example/README.md:50]: Flutter also has docs on signing apks here https://docs.flutter.dev/deployment/android#signing-the-app

File: packages/in_app_purchase/in_app_purchase_android/lib/src/in_app_purchase_android_platform_addition.dart (hunk 1 above)
Comment (gmackall) [reply packages/in_app_purchase/in_app_purchase_android/lib/src/in_app_purchase_android_platform_addition.dart:155]: I think this comment still applies

File: packages/in_app_purchase/in_app_purchase_android/lib/src/in_app_purchase_android_platform_addition.dart [hunk 5]
```diff
@@ -146,4 +146,15 @@ class InAppPurchaseAndroidPlatformAddition
       (BillingClient client) => client.isFeatureSupported(feature),
     );
   }
+
+  /// Returns Play billing country code based on ISO-3166-1 alpha2 format.
+  ///
+  /// See: https://developer.android.com/reference/com/android/billingclient/api/BillingConfig
+  /// See: https://unicode.org/cldr/charts/latest/supplemental/territory_containment_un_m_49.html
+  Future<String> getCountryCode() async {
```
Comment (gmackall) [inline packages/in_app_purchase/in_app_purchase_android/lib/src/in_app_purchase_android_platform_addition.dart:154]: I think method should get tests added to `in_app_purchase_android_platform_addition_test.dart` to cover

File: packages/in_app_purchase/in_app_purchase_storekit/example/README.md [hunk 6]
```diff
@@ -7,3 +7,75 @@ package.
...
+# In App Purchase iOS Example
+
+### Preparation
+
+There's a significant amount of setup required for testing in app purchases
+successfully, including registering new app IDs and store entries to use for
+testing in App Store Connect. The App Store requires developers to configure
+an app with in-app items for purchase to call their in-app-purchase APIs.
+The App Store has extensive documentation on how to do this, and we've also
+included a high level guide below.
```
Comment (jmagman) [inline packages/in_app_purchase/in_app_purchase_storekit/example/README.md:20]: Can we avoid adding a high level guide?  These things go out of date so quickly, and we don't do a good job of keeping track of when processes like this change. I'd rather just link to the Apple docs.  If we need something for our own reference for testing I'd rather it go into the wiki.

File: packages/in_app_purchase/in_app_purchase_storekit/example/README.md [hunk 7]
```diff
@@ -7,3 +7,75 @@ package.
...
+
+   For step #2, "Configure in-app purchases in App Store Connect," you'll want
+   to create the following products:
+
+   - A consumable with product ID `consumable`
+   - An upgrade with product ID `upgrade`
+   - An auto-renewing subscription with product ID `subscription_silver`
+   - An non-renewing subscription with product ID `subscription_gold`
+
+2. In XCode, `File > Open File` `example/ios/Runner.xcworkspace`. Update the
```
Comment (jmagman) [inline packages/in_app_purchase/in_app_purchase_storekit/example/README.md:62]:
```suggestion
2. In Xcode, `File > Open File` `example/ios/Runner.xcworkspace`. Update the
```

File: packages/in_app_purchase/in_app_purchase_storekit/example/README.md (hunk 6 above)
Comment (reidbaker) [reply packages/in_app_purchase/in_app_purchase_storekit/example/README.md:20]: This is a revert from https://github.com/flutter/packages/commit/bdb89758e11767c0e9770b46aee72ffe0301b3c1#diff-896f125f477201f602e98fef2399dfb4c17297b851fa7df48bd608a689f3c542

Without these instructions and with this plugin being maintained by a contract team but that contract ending I had zero idea how to test the example app. 

I can revert the ios changes but the android example app explicitly references the high level guide so android will keep its guide.

**Review by gmackall — APPROVED** [review]
LGTM outside of the format/analyze failures!

**Pull request merged**